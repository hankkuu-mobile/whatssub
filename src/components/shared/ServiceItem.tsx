import React, { Component, useRef } from 'react';
import { TouchableOpacity, View } from 'react-native';

import { IC_LIST_BTN } from '../../utils/Icons';
import { colors } from '../../theme';
import styled from 'styled-components/native';

const Container = styled.View`
  height: 50px;
  width: 100%;
  background-color: ${colors.headerBackgroundLight};
  flex-direction: row;
  align-items: center;
  padding: 0 16px;
`;

const Title = styled.Text`
  color: ${colors.textPrimaryLight};
  font-size: 16px;
  font-family: 'spoqa-han-sans-regular';
`;

const Content = styled.TextInput`
  color: ${colors.textSecondaryLight};
  font-size: 16px;
  font-family: 'spoqa-han-sans-regular';
`;

const Arrow = styled.Image`
  width: 8px;
  height: 13px;
  margin-left: 11px;
`;

interface Props {
  title: string;
  content: string;
  testID?: string;
  contentStyle?: any;
  showArrow?: boolean;
  onPress?(): any;
  editable?: boolean;
  onChangeContent?(string): void;
}

function ServiceItem(props: Props) {
  const {
    title = '제목',
    content = '내용',
    contentStyle,
    showArrow,
    editable = false,
    onPress = null,
    onChangeContent = null,
  } = props;
  const inputText = useRef(null);
  return (
    <TouchableOpacity
      testID={props.testID}
      onPress={
        // prettier-ignore
        props.onPress
          ? props.onPress
          : editable && inputText != null
            ? () => inputText.current.focus()
            : null
      }
      activeOpacity={props.onPress ? 0.2 : 1}
    >
      <Container>
        <Title>{title}</Title>
        <View style={{ flex: 1 }} />
        <Content
          ref={inputText}
          style={contentStyle}
          editable={editable}
          pointerEvents={editable ? 'auto' : 'none'}
          onChangeText={onChangeContent}
        >
          {content}
        </Content>
        {showArrow ? <Arrow source={IC_LIST_BTN} /> : null}
      </Container>
    </TouchableOpacity>
  );
}

export default ServiceItem;
