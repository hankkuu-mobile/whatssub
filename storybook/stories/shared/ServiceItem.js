import React from 'react';
import ServiceItem from '../../../src/components/shared/ServiceItem';
import { colors } from '../../../src/theme';
import { storiesOf } from '@storybook/react-native';
import styled from 'styled-components/native';

const Container = styled.View`
  flex: 1;
  justify-content: center;
  background-color: ${colors.gray};
`;

const Description = styled.Text`
  padding: 10px;
`;

const Story = () => {
  const [editableContent, setEditableContent] = React.useState('내용');
  return (
    <Container>
      <Description>Default</Description>
      <ServiceItem/>
      <Description>Clickable</Description>
      <ServiceItem title='클릭가능한 아이템' onPress={() => { /* DO SOMTHING */ }}/>
      <Description>Editable {editableContent}</Description>
      <ServiceItem
        title='편집할 수 있는 아이템'
        editable
        onChangeContent={(value) => setEditableContent(value)}/>
    </Container>
  );
};

storiesOf('shared-ServiceItem', module)
  .add('default', () => <Story />);
