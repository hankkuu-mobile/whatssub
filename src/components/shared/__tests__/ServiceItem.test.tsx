import 'react-native';
import * as React from 'react';
import { RenderResult, act, fireEvent, render } from '@testing-library/react-native';
import ServiceItem from '../ServiceItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

const createTestProps = (obj: object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...obj,
});

describe('[ServiceItem] render', () => {
  beforeEach(() => {
    props = createTestProps({ });
    component = (
      <ServiceItem {...props} />
    );
  });

  it('renders without crashing', () => {
    const rendered: renderer.ReactTestRendererJSON = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('interactions', () => {
    beforeEach(() => {
      testingLib = render(component);
    });

    it('should simulate onClick', () => {
      // const btn = testingLib.queryByTestId('btn');
      // act(() => {
      //   fireEvent.press(btn);
      // });
      // expect(cnt).toBe(3);
    });
  });
});
