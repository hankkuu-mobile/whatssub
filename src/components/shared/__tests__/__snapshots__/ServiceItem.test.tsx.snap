// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ServiceItem] render renders without crashing 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "height": 50,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#323b43",
            "fontFamily": "spoqa-han-sans-regular",
            "fontSize": 16,
          },
        ]
      }
    >
      제목
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      editable={false}
      onChangeText={null}
      pointerEvents="none"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "#99a2aa",
            "fontFamily": "spoqa-han-sans-regular",
            "fontSize": 16,
          },
        ]
      }
      underlineColorAndroid="transparent"
    >
      내용
    </TextInput>
  </View>
</View>
`;
